deeplab/demo/src/index.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import 'bulma/css/bulma.css';
+import { SemanticSegmentation } from '@tensorflow-models/deeplab';
+import pascalExampleImage from './examples/pascal.jpg';
+import cityscapesExampleImage from './examples/cityscapes.jpg';
+import ade20kExampleImage from './examples/ade20k.jpg';
+
+const state = {
+  isQuantized: true,
+};
+
+const deeplab = {
+  pascal: undefined,
+  cityscapes: undefined,
+  ade20k: undefined,
+};
+
+const deeplabExampleImages = {
+  pascal: pascalExampleImage,
+  cityscapes: cityscapesExampleImage,
+  ade20k: ade20kExampleImage,
+};
+
+const initialiseModels = async () => {
+  Object.keys(deeplab).forEach(modelName => {
+    if (deeplab[modelName]) {
+      deeplab[modelName].dispose();
+    }
+    const model = new SemanticSegmentation(modelName, state.isQuantized);
+    deeplab[modelName] = model;
+    const toggler = document.getElementById(`toggle-${modelName}-image`);
+    toggler.onclick = () => setImage(deeplabExampleImages[modelName]);
+    const runner = document.getElementById(`run-${modelName}`);
+    runner.onclick = async () => {
+      runner.classList.add('is-loading');
+      await sleep(100);
```

Comment:
Do we need to do this?

Reply:
This allows the main thread to animate the loading state of the button. Let me know if there is a better way to do this.

Reply:
I think it can be solved using [`tf.nextFrame()`](https://js.tensorflow.org/api/latest/#nextFrame). Have you explored it?